### DISCUSSION TITLE
Will the remark42 docker image work in a rootless docker environment?

### DISCUSSION THREAD

**Discussion by georgew509** [umputun/remark42#1852] [Q&A] [answered]
My VPS provider notes that "Our Docker installations run in a rootless environment, which may limit certain applications." I am new to remark42 and Docker, so I wanted to confirm that remark42 should work in this environment. If so, it would be great if someone could point me to a good tutorial for setting up remark42. My website is using the Hugo static site generator and runs on the above mentioned VPS that uses the Apache web server. I do not have any root access. Thanks!

**Comment by paskal** [comment]
It's working under non-root user 1001 by default, you shouldn't have any problems with that: https://remark42.com/docs/configuration/parameters/#docker-image

You can alter the id of that user in case your provider has a particular preference. I think any tutorial will do, https://remark42.com/docs/getting-started/installation/ is a good starting point. Please let me know if something is missing there.

**Comment by georgew509** [comment]
Thanks for the pointers. I tried following the installation instructions, but it doesn’t seem to be working.

If you don’t mind, I’ll share a few details (minus any client secrets):

At the root of my Remark42 web server:
… public_html/comments.altoplace.com <http://comments.altoplace.com/>

$ l
docker-compose.yml  var/

$ cat docker-compose.yml
services:
  remark42:
    image: umputun/remark42:latest
    restart: always
    container_name: "remark42"
    environment:
      - REMARK_URL=https://comments.altoplace.com # URL pointing to your Remark42 server
      - SITE=remark42 # site ID, same as used for `site_id`, see "Setup on your website"
      - SECRET=xxxxxxxxxx # secret key
      - AUTH_ANON=true # enable anonymous commenting
      - AUTH_GITHUB_CID=xxxxxxxxxxxxx # OAuth2 client ID
      - AUTH_GITHUB_CSEC=yyyyyyyyyyyyyy # OAuth2 client secret
    volumes:
      - ./var:/srv/var # persistent volume to store all Remark42 data

I did:

docker compose pull && docker compose up -d

Seems to be running:

$ docker ps -a
CONTAINER ID   IMAGE                     COMMAND                  CREATED          STATUS                    PORTS      NAMES
674c21a20f77   umputun/remark42:latest   "/init.sh /srv/remar…"   16 minutes ago   Up 16 minutes (healthy)   8080/tcp   remark42

But the following tests I found are failing:

$ curl http://127.0.0.1:8080
curl: (7) Failed to connect to 127.0.0.1 port 8080 after 0 ms: Could not connect to server

I set up a reverse proxy for my remark42 URL and tried:
https://comments.altoplace.com/web

And I get a 503 service unavailable.

Any ideas about what I could be doing wrong? How to verify my docker environment?

Oh, btw, I deleted the 'version: “2”’ line because I saw a warning that said “version” is obsolete and no longer used.

Thanks,
George

**Reply by paskal** [reply, marked as answer]
You should be able to connect to `127.0.0.1:8080`, but you are not exposing any ports from the container (`ports:` section in the example docker-compose.yml) so you have no ability to connect to the container. Same goes for reverse proxy, first you need to verify that whatever address reverse proxy is proxying the request to, is accessible to you using `curl` from the same machine.

In your scenario with reverse proxy, I would recommend something like this:

```
    ports:
      - "127.0.0.1:8080:8080"
```

That way it would be open for local test with curl and for reverse proxy, but it would not be possible to reach your Remark42 instance by IP.

**Comment by georgew509** [comment]
Ok, curl and the reverse proxy URL are now working. I had missed that detail.

Thanks much for the help.
